addons/pos_self_order/controllers/controllers.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+import werkzeug
+from odoo import http
+from odoo.http import request
+from typing import List, Dict
+from odoo.addons.http_routing.models.ir_http import unslug
+
+
+class PosSelfOrderController(http.Controller):
+    """
+    This is the controller for the POS Self Order App
+    There is one main route that the client will use to access the POS Self Order App: /menu
+    This route will render the LANDING PAGE of the POS Self Order App
+    And it will pass some generic variables to the template: pos_config_id, table_id, company_name, currency...
+    After that the client will be able to navigate to the /products route w/o aditional requests
+    to the server, using client side routing.
+    """
+
+    @http.route("/menu", auth="public")
+    def pos_self_order_redirect(self):
+        pos_config_id = self._get_any_pos_config_with_qr_code_menu_sudo().id
+        return request.redirect(f"/menu/{pos_config_id}")
+
+    @http.route(
+        [
+            "/menu/<string:pos_name>",
+            "/menu/<string:pos_name>/products",
+            "/menu/<string:pos_name>/products/<int:product_id>",
+        ],
+        auth="public",
+        website=True,
+        sitemap=True
+    )
+    def pos_self_order_start(self, pos_name: str):
+        """
+        The user gets this route from the QR code that they scan at the table
+        This START ROUTE will render the POS Self Order App
+        And it will pass the needed data to the template: the product list, pos_config_id, company_name, currency, etc.
+        :param pos_name: the name of the pos config: can be the id or the slugified name of the pos config
+        :type pos_name: str
+        :return: the rendered template
+        """
+        _, pos_config_id = unslug(pos_name)
+        pos_config_sudo = self._get_pos_config_sudo(pos_config_id)
+        session_info = request.env["ir.http"].get_frontend_session_info()
+        session_info["currencies"] = request.env["ir.http"].get_currencies()
+        session_info["pos_self_order"] = self._get_self_order_config(
+            pos_config_sudo)
+        response = request.render(
+            "pos_self_order.index",
+            {
+                "session_info": session_info,
```

Comment:
`session_info` is a bit of a misnomer, it's a bag of data that's needed to start the application and for which we don't want to make separate requests. Historically it's been mostly session-related stuff hence the name but there are quite a few things in there that aren't really session related, even in the web client.

Because there is an easy/standard mechanism to access the data in `odoo.__session_info__` in js (importing the `@web/session` module) and it avoids further polluting the global scope or the `odoo` global I suggested that he use this.

I've talked about this with the JS team, and we may want to rename this eventually, and have actual session-related stuff in a session field in that renamed object instead, but the rest of the team thinks that it's premature/overkill for now.